<launch>
	<!-- world name -->
	<arg name="world_name" default="$(find ardupilot_sitl_gazebo_plugin)/worlds/empty_world/empty.world"/>
	<!-- model name -->
	<!--<arg name="model" default="$(find auto_takeoff)/urdf/erlecopter_with_wind_gust.xacro"/>-->
	<arg name="model" default="$(find auto_takeoff)/urdf/test.xacro"/>

	<!-- other args and params -->
	<arg name="simRate" default="nan"/>
	<!-- Enable simulation clock -->
	<arg name="enable_logging" default="true"/>
	<arg name="enable_ground_truth" default="true"/>
	<arg name="log_file" default="erlecopter"/>
	<arg name="headless" default="false"/>
	<arg name="gui" default="true"/>
	<arg name="name" default="erlecopter"/>
	<arg name="tf_prefix" default="$(optenv ROS_NAMESPACE)"/>
	<arg name="debug" default="true"/>
	<arg name="verbose" default="true"/>

	<env name="GAZEBO_MODEL_PATH" value="$(find drcsim_model_resources)/gazebo_models/environments:$(find ardupilot_sitl_gazebo_plugin)/meshes/meshes_sensors:$(find ardupilot_sitl_gazebo_plugin)/meshes/meshes_outdoor:$(find ardupilot_sitl_gazebo_plugin)/meshes/meshes_warehouse"/>

	<param name="tf_prefix" type="string" value="$(arg tf_prefix)" />
	<param name="use_sim_time" type="bool" value="true" />

	<include file="$(find mavros)/launch/apm_sitl.launch"/>

	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="paused" value="true"/>   <!-- value irrelevant due to Arducopter plugin steps -->
		<arg name="headless" value="$(arg headless)"/>
		<arg name="gui" value="$(arg gui)"/>
		<arg name="world_name" value="$(arg world_name)"/>
		<arg name="verbose" value="$(arg verbose)"/>
	</include>

	<include file="$(find auto_takeoff)/launch/spawn_erlecopter.launch">
		<arg name="model" value="$(arg model)"/>
		<arg name="enable_logging" value="$(arg enable_logging)" />
		<arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
		<arg name="log_file" value="$(arg log_file)"/>
	</include>

	<node name="auto_takeoff" pkg="auto_takeoff" type="auto_takeoff" output="screen"></node>
	<node name="wind_publisher" pkg="auto_takeoff" type="wind_publisher" output="screen"></node>

	<!--  <node name="tag1" pkg="gazebo_ros" type="spawn_model" args="-file $(find rotors_description)/urdf/ARtag.urdf -urdf -x 0.0 -y 0.0 -z 0.01 -model ARtag1" respawn="false" />-->

</launch>
